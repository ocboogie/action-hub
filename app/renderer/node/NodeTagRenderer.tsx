import { TagRendererProps } from "cheerio-react-bind";
import * as React from "react";

import { IRegistry } from "../registry";
import { INode } from "./index";
import NodeRenderer from "./NodeRenderer";

interface INodeTagRendererProps extends TagRendererProps {
  registry: IRegistry;
}

export default class NodeTagRenderer extends React.Component<
  INodeTagRendererProps
> {
  public node: INode;

  constructor(props: INodeTagRendererProps) {
    super(props);

    this.node = this.props.registry.nodes[this.props.tagName];
    if (!this.node) {
      // tslint:disable-next-line: no-string-throw
      throw `Unknown node name "${this.props.tagName}".`;
    }
  }

  public render() {
    if (!this.node) {
      return <div />;
    }

    return (
      <>
        <NodeRenderer args={this.props.attributes} node={this.node}>
          {this.props.children}
        </NodeRenderer>
      </>
    );
  }
}
