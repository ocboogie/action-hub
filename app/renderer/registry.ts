import { INode } from "./node";

export interface IRegistry {
  nodes: {
    [nodeName: string]: INode;
  };
}
