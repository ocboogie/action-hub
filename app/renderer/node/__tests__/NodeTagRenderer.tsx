// tslint:disable-next-line:no-implicit-dependencies
import { mount, shallow } from "enzyme";
import * as React from "react";

import NodeTagRenderer from "../NodeTagRenderer";
import registry from "./resources/mockRegistry";

test("renders node", () => {
  const wrapper = mount(
    <NodeTagRenderer
      location="/"
      tagName="text"
      attributes={{ text: "foo" }}
      registry={registry}
    />
  );

  expect(wrapper.contains(<h1>foo</h1>)).toBeTruthy();
});

test("logs when unknown node name is found", () => {
  expect(() => {
    shallow(
      <NodeTagRenderer
        location="/"
        tagName="bar?"
        attributes={{}}
        registry={registry}
      />
    );
  }).toThrowErrorMatchingSnapshot();
});
