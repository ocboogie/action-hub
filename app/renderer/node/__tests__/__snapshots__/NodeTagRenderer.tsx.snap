// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logs when unknown node name is found 1`] = `undefined`;
